File: client_leger/lib/backend-communication-services/models/chat_channels.dart
class ChatChannel {
  final String name;
  final bool isUserInChannel;

  ChatChannel({required this.name, required this.isUserInChannel});

  factory ChatChannel.fromJson(
      Map<String, dynamic> json, String currentUserId) {
    final users = List<String>.from(json['users'] ?? []);
    final isUserInChannel = users.contains(currentUserId);
    return ChatChannel(
      name: json['name'] as String,
      isUserInChannel: isUserInChannel,
    );
  }
}

File: client_lourd/src/app/services/chat-services/chat-channels.ts
export interface ChatChannel {
    name: string;
    isUserInChannel: boolean;
}

export function chatChannelFromJson(json: any, currentUserId: string): ChatChannel {
    const users: string[] = json.users || [];
    const isUserInChannel = users.includes(currentUserId);
    return {
        name: json.name,
        isUserInChannel: isUserInChannel,
    };
}